src/foam/u2/DateView.js

```diff
@@ -34,19 +38,36 @@ foam.CLASS({
     function initE() {
       this.SUPER();
       this.setAttribute('type', 'date');
-      this.setAttribute('placeholder', 'yyyy/mm/dd');
+      this.on('change', () => {
+        this.checkDateFormat(this.getAttribute('value'));
+      });
     },
 
     function link() {
       this.data$.relateTo(
           this.attrSlot(null, this.onKey ? 'input' : null),
           function(date) {
-            return date ? date.toISOString().substring(0,10) : date;
+            return date ? date.toISOString().substring(0, 10) : date;
           },
           function(value) {
             return new Date(value);
           }
       );
     }
+  ],
+
+  listeners: [
+    function checkDateFormat(value) {
+      oldDate = value.split(/[^0-9]/).map((str) => {
+        return parseInt(str, 10);
+      }).join('');
+
+      var newDate = new Date(oldDate);
+      newDate = `${newDate.getFullYear()}${newDate.getMonth() + 1}${newDate.getDate()}`;
+      if ( newDate !== oldDate ) {
+        ctrl.notify('Invalid Date.', 'error');
```

Comment:
Please remove the notify and replace with a console error.

Reply:
Console.error won't be a good way. We need a way to notify the user their input is not appropriate